lib/fastlane/plugin/ddg_apple_automation/actions/asana_add_comment_action.rb

```diff
@@ -0,0 +1,130 @@
+require "fastlane/action"
+require "fastlane_core/configuration/config_item"
+require "asana"
+require_relative "../helper/ddg_apple_automation_helper"
+require_relative "asana_extract_task_id_action"
+
+module Fastlane
+  module Actions
+    class AsanaAddCommentAction < Action
+      def self.run(params)
+        asana_access_token = params[:asana_access_token]
+        task_id = params[:task_id]
+        task_url = params[:task_url]
+        template_name = params[:template_name]
+        comment = params[:comment]
+        workflow_url = params[:workflow_url]
+
+        begin
+          validate_params(task_id, task_url, comment, template_name, workflow_url)
+        rescue ArgumentError => e
+          UI.user_error!(e.message)
+          return
+        end
+
+        task_id = Fastlane::Actions::AsanaExtractTaskIdAction.run(task_url: task_url) if task_url
+
+        if template_name
+          template_content = load_template_file(template_name)
+          return unless template_content
```

Comment:
Let's show `user_error!` when template is not found

Reply:
I stand corrected, just realized that load_template_file shows error and here we're just returning to make RSpec happy 👍